scripts_ZA/compareShapesForTTbarEstim.py

```diff
@@ -0,0 +1,298 @@
+
+    list_histos_data = addHistos(path_data, True)
+    list_histos_ttbarMC = addHistos(path_MC, False) 
+    list_histos_singleTop = getSingleTop(path_MC)
+
+    legend = []
+    c1 = []
+    c2 = []
+    c3 = []
+    pad1 = []
+    pad2 = []
+    subtractSingleTop = False
+
+    for i in range(0, len(list_histos_data)):
+
+        norm_data = list_histos_data[i].Integral()
+        norm_ttbar = list_histos_ttbarMC[i].Integral()
+        norm_singletop = list_histos_singleTop[i].Integral()
+
+        print norm_data
+        print norm_ttbar
+        print norm_singletop
+
+
+        c1.append(TCanvas("c1","c1",600,600))
+        pad1.append(TPad("pad1","pad1",0,0.3,1,0.9))
+        pad1[i].SetBottomMargin(0)
+        pad1[i].Draw()
+        pad1[i].cd()
+        list_histos_data[i].SetStats(0)
+        list_histos_data[i].SetMarkerColor(ROOT.kRed)
+        list_histos_data[i].SetMarkerStyle(20)
+        list_histos_ttbarMC[i].SetMarkerColor(ROOT.kBlue)
+        list_histos_ttbarMC[i].SetMarkerStyle(22)
+        # Subtract the SingleTop background
+        if subtractSingleTop:
+            list_histos_data[i].Add(list_histos_singleTop[i], -1)
+        list_histos_data[i].Scale(1/norm_data)
+        list_histos_data[i].Draw("")
+        list_histos_ttbarMC[i].Scale(1/norm_ttbar)
+        list_histos_ttbarMC[i].Draw("same")
+        legend.append(ROOT.TLegend(0.5, 0.7, 0.8, 0.8))
+        legend[i].AddEntry(list_histos_data[i], "data (El-Mu category)", "p")
+        legend[i].AddEntry(list_histos_ttbarMC[i], "MC ttbar (Mu-Mu category)", "p")
+        legend[i].Draw("same")
+
+        c1[i].cd()
+        pad2.append(TPad("pad2","pad2",0, 0.05, 1, 0.3))
+        pad2[i].SetTopMargin(0)
+        pad2[i].SetBottomMargin(0.3)
+        pad2[i].Draw()
+        pad2[i].cd()
+        ratio = list_histos_data[i].Clone("Ratio")
+        #ROOT.gStyle.SetOptTitle(0)
+        ratio.SetTitle("")
+        ratio.Sumw2()
+        ratio.Divide(list_histos_ttbarMC[i])
+        ratio.SetMarkerColor(ROOT.kBlack)
+        ratio.SetMarkerSize(0.8)
+        ratio.SetStats(0)
+        ratio.GetYaxis().SetRangeUser(0,2)
+        ratio.Draw("same")
+        line = TLine(ratio.GetXaxis().GetXmin(), 1, ratio.GetXaxis().GetXmax(), 1)
+        line.SetLineColor(ROOT.kBlack)
+        line.Draw("")
+
+        c1[i].cd()
+        if not subtractSingleTop:
+            c1[i].SaveAs("compareShapesForTTbar/plot_%i.png" % i)
+            #c1[i].SaveAs("compareShapesForTTbar/plot_%i.pdf" % i)
+        elif subtractSingleTop:
+            c1[i].SaveAs("compareShapesForTTbar/plot_subtractSingleTop%i.png" % i)
+            #c1[i].SaveAs("compareShapesForTTbar/plot_%i.pdf" % i)
+            
+
+        c3.append(TCanvas("c3","c3",600,600))
+        c3[i].cd()
+        list_histos_singleTop[i].Draw("")
+        c3[i].SaveAs("compareShapesForTTbar/plot_SingleTop_%i.png" % i)
```

Comment:
Is it possible to see all these plots ? I'm curious :smile:

Reply:
They're here: /nfs/scratch/fynu/asaggio/CMSSW_8_0_30/src/cp3_llbb/ZATools/scripts_ZA/compareShapesForTTbar :)

Reply:
nice !